Detections/ASimDNS/imDns_DomainEntity_DnsEvents.yaml

```diff
@@ -0,0 +1,66 @@
+id: 999e9f5d-db4a-4b07-a206-29c4e667b7e8
+name: (Preview) TI map Domain entity to Dns Event (Normalized DNS)
+description: |
+  'Identifies a match in DNS events from any Domain IOC from TI
+  To use this analytics rule, make sure you have deployed the [ASIM normalization parsers](https://aka.ms/AzSentinelDns)'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: ThreatIntelligence
+    dataTypes:
+      - ThreatIntelligenceIndicator
+  - connectorId: ThreatIntelligenceTaxii
+    dataTypes:
+      - ThreatIntelligenceIndicator
+queryFrequency: 1h
+queryPeriod: 14d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Impact
+tags:
+  - Id: 85aca4d1-5d15-4001-abd9-acb86ca1786a
+    version: 1.0.0
+  - Schema: ASIMDns
+    SchemaVersion: 0.1.1
+query: |
+  let dt_lookBack = 1h;
+  let ioc_lookBack = 14d;
+  //Create a list of TLDs in our threat feed for later validation
+  let list_tlds = ThreatIntelligenceIndicator
+    | where TimeGenerated > ago(ioc_lookBack)
+    | where isnotempty(DomainName)
+    | extend tld = split(DomainName, '.')[-1]
+    | summarize make_set(tld);
+  ThreatIntelligenceIndicator
+    | where TimeGenerated >= ago(ioc_lookBack) and ExpirationDateTime > now()
+    | where Active == true
+    // Picking up only IOC's that contain the entities we want
+    | where isnotempty(DomainName)
+    | join (
+        imDns(starttime=ago(dt_lookBack))
+        //Extract domain patterns from syslog message
+        | where isnotempty(DnsQuery)
+        //Split out the TLD
+        | extend parts = split(DnsQuery, '.')
+        //Validate parsed domain by checking if the TLD is in the list of TLDs in our threat feed
+        | extend tld = split(DnsQuery, '.')[-1]
+        | extend DNS_TimeGenerated = TimeGenerated
+  ) on $left.DomainName==$right.DnsQuery
+  | where DNS_TimeGenerated >= TimeGenerated and DNS_TimeGenerated < ExpirationDateTime
```

Comment:
Why repeat the "TimeGenerated" test part (i.e. the first test)?

Reply:
It is not the same TimeGenerated. This one comes from the TI table, and line 49 verifies that the event timestamp  (DNS_TimeGenerated) is within the validity  timeframe of the TI: after it was marked as suspicious and before it expired

Reply:
Actually, it is a bad bug in the original query. Since the refresh the table every 14 days, TimeGenerated is not a correct value to compare to for the "start of life" of the IoC. They have a bug, that in the refresh they don't maintain the original TimeGenerated in any field, however, in practice. If there is a new IoC and an old event, you still want to alert on it, so I would remove this check.